src/util/entities/Channel.ts

```diff
@@ -243,6 +262,100 @@ export class Channel extends BaseClass {
 		return channel;
 	}
 
+	static async createThreadChannel(
+		channel: Partial<Channel>,
+		metadata: Partial<ThreadMetadataSchema>,
+		user_id: string = "0",
+		opts?: {
+			keepId?: boolean;
+			skipExistsCheck?: boolean;
+			skipParentExistsCheck?: boolean;
+			skipPermissionCheck?: boolean;
+			skipEventEmit?: boolean;
+			skipNameChecks?: boolean;
+		}
+	): Promise<Channel> {
+		channel = {
+			// set the default type to private
+			type: ChannelType.GUILD_PRIVATE_THREAD,
+			...channel,
+			...(!opts?.keepId && { id: Snowflake.generate() }),
+			created_at: new Date(),
+			position: 0, // TODO:
+			message_count: 0,
+			member_count: 1,
+			total_message_sent: 0
+		};
+
+		const exists = await Channel.findOne({
+			where: {
+				id: channel.id
+			}
+		});
+		if (!opts?.skipExistsCheck && exists) throw DiscordApiErrors.THREAD_ALREADY_CREATED_FOR_THIS_MESSAGE;
+
+		if (!channel.parent_id) throw new HTTPError("Parent id not set", 400);
+		const parent = await Channel.findOneOrFail({ where: { id: channel.parent_id } });
+
+		if (!opts?.skipPermissionCheck) {
+			// Always check if user has permission first
+			const permissions = await getPermission(user_id, parent.guild_id);
+			permissions.hasThrow(channel.type === ChannelType.GUILD_PUBLIC_THREAD ? "CREATE_PUBLIC_THREADS" : "CREATE_PRIVATE_THREADS");
+		}
+
+		channel = {
+			...channel,
+			permission_overwrites: parent.permission_overwrites,
+			nsfw: parent.nsfw,
+			owner_id: user_id,
+			guild_id: parent.guild_id,
+			thread_metadata: {
+				create_timestamp: new Date(),
+				archive_timestamp: new Date(),
+				archived: false,
+				auto_archive_duration: 0,
+				invitable: channel.type === ChannelType.GUILD_PRIVATE_THREAD ? false : true,
```

Comment:
default for invitable in public threads should also be configurable

Reply:
see the latest review note: AAD=0 was incompatible with the original type of AAD of {60, 1440, 4320, 10080}.